gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2

```diff
@@ -93,14 +104,29 @@ class Async{{service.name}}RestTransport(_Base{{ service.name }}RestTransport):
     {{ shared_macros.wrap_async_method_macro()|indent(4) }}
 
     {% for method in service.methods.values()|sort(attribute="name") %}
+    class _{{method.name}}(_Base{{ service.name }}RestTransport._Base{{method.name}}, Async{{service.name}}RestStub):
+        def __hash__(self):
+            return hash("Async{{service.name}}RestTransport.{{method.name}}")
+
+        async def __call__(self,
+                    request: {{method.input.ident}}, *,
+                    retry: OptionalRetry=gapic_v1.method.DEFAULT,
+                    timeout: Optional[float]=None,
+                    metadata: Sequence[Tuple[str, str]]=(),
+                    {# TODO(b/362949446): Update the return type as we implement this for different method types. #}
+                    ) -> None:
+                raise NotImplementedError(
+                    "Method {{ method.name }} is not available over REST transport"
+                )
 
-    {# TODO(b/362949446) Return a callable once implemented and remove `type:  ignore`. #}
+    {% endfor %}
+    {% for method in service.methods.values()|sort(attribute="name") %}
     @property
     def {{method.transport_safe_name|snake_case}}(self) -> Callable[
             [{{method.input.ident}}],
             {{method.output.ident}}]:
         
-        return  # type: ignore
+        return self._{{method.name}}(self._session, self._host)  # type: ignore
```

Comment:
Do we still need the `# type: ignore` ?

Reply:
Yes. My initial thought was that it's probably a mismatch since I wasn't returning a Callable. However, I later realised that we're doing the same in `rest.py`.

I haven't really looked into why we've been ignoring the type in `rest.py` but I'm following whatever we're doing there. Probably because the Callables for client streaming which raise `NotImplmentedError` do not return anything so mypy isn't happy about it? Regardless, I've filed an issue to investigate and remove this as a follow up.